File: java-samples/Advanced/duediligence-cordapp/contracts/src/main/java/net/corda/samples/duediligence/contracts/DueDChecklistContract.java
package net.corda.samples.duediligence.contracts;

import net.corda.core.contracts.*;
import net.corda.core.transactions.LedgerTransaction;
import net.corda.samples.duediligence.states.CopyOfCoporateRecordsAuditRequest;
import net.corda.samples.duediligence.states.CorporateRecordsAuditRequest;
import org.jetbrains.annotations.NotNull;

import java.util.List;

import static net.corda.core.contracts.ContractsDSL.requireThat;

public class DueDChecklistContract implements Contract {
    @Override
    public void verify(@NotNull LedgerTransaction tx) throws IllegalArgumentException {
        final CommandWithParties command = tx.getCommands().get(0);

        //Propose request
        if (command.getValue() instanceof DueDChecklistContract.Commands.Add) {
            requireThat(require -> {
                StateAndRef<ContractState> input = tx.getInputs().get(0);
                if (input.getState().getData().getClass() == CorporateRecordsAuditRequest.class){
                    CorporateRecordsAuditRequest request = (CorporateRecordsAuditRequest)input.getState().getData();
                    require.using("Qualification Must be True", request.getQualification());
                }else{
                    CopyOfCoporateRecordsAuditRequest request = (CopyOfCoporateRecordsAuditRequest) input.getState().getData();
                    require.using("Qualification Must be True", request.getQualification());
                }
                return null;
            });
        }
    }

    public interface Commands extends CommandData {
        class Add implements Commands { }
    }
}

File: kotlin-samples/Advanced/duediligence-cordapp/contracts/src/main/kotlin/net/corda/samples/duediligence/contracts/DueDChecklistContract.kt
package net.corda.samples.duediligence.contracts

import net.corda.core.contracts.*
import net.corda.core.contracts.Requirements.using
import net.corda.core.transactions.LedgerTransaction
import net.corda.samples.duediligence.states.CopyOfCoporateRecordsAuditRequest
import net.corda.samples.duediligence.states.CorporateRecordsAuditRequest
import java.lang.IllegalArgumentException

class DueDChecklistContract :Contract{

    companion object {
        // Used to identify our contract when building a transaction.
        const val ID = "net.corda.samples.duediligence.contracts.DueDChecklistContract"
    }

    @Throws(IllegalArgumentException::class)
    override fun verify(tx: LedgerTransaction) {
        val command = tx.commands[0]

        //Propose request
        when (command.value){
            is Commands.Add -> requireThat {
                val (state) = tx.inputs[0]
                if (state.data.javaClass == CorporateRecordsAuditRequest::class.java) {
                    val request = state.data as CorporateRecordsAuditRequest
                    "Qualification Must be True".using(request.qualification)
                } else {
                    val request = state.data as CopyOfCoporateRecordsAuditRequest
                    "Qualification Must be True".using(request.qualification)
                }
            }
        }
    }

    interface Commands : CommandData {
        class Add : Commands
    }
}